nucypher/characters/control/interfaces.py

```diff
@@ -220,6 +221,19 @@ def retrieve(self,
 
         self.character.join_policy(label=label, publisher_verifying_key=alice_verifying_key)
 
+        if self.character.federated_only:
+            from nucypher.policy.maps import TreasureMap as _MapClass
+        else:
+            from nucypher.policy.maps import SignedTreasureMap as _MapClass
+
+        # TODO: This LBYL is ugly and fraught with danger.  NRN
+        if isinstance(treasure_map, bytes):
```

Comment:
I'm a tad bit confused here, the `/retrieve` endpoint schema `Bob.Retrieve` (https://github.com/fjarri/nucypher/blob/retrieval-protocol/nucypher/characters/control/specifications/bob.py#L43) does not include a treasure map in the request data - how is a treasure map even passed here via a request? Presumably, it is always `None` (L207).

Pinging @vepkenez / @jMyles, since the treasure_map parameter in this function is prior to @fjarri 's current change.

Reply:
Hmm, yeah, I'm also not sure.  It looks like it was added when @michwill crafted the 'bring your own map' logic, but then we never went the extra step of making any of our controllers pass it.

Reply:
Fixed in #2768.